Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -858,6 +939,59 @@ AXTextMarkerRange AXTextMarker::lineRange(LineRangeType type) const
     RELEASE_ASSERT_NOT_REACHED();
 }
 
+AXTextMarkerRange AXTextMarker::wordRange(WordRangeType type) const
+{
+    if (!isValid())
+        return { { }, { } };
+
+    AXTextMarker startMarker, endMarker;
+
+    if (type == WordRangeType::Right) {
+        startMarker = nextWordStart();
+        endMarker = startMarker.nextWordEnd();
+
+        if (startMarker == *this) {
+            auto resultRange = AXTextMarkerRange { startMarker, endMarker };
+            auto resultRangeString = resultRange.toString();
+            // A newline should not be returned as the right word (if we are at the end of a paragraph)
+            if (resultRangeString.length() && resultRangeString[0] == '\n')
+                return { WTFMove(*this), WTFMove(*this) };
+        }
+    } else {
+        startMarker = previousWordStart();
+        // If the current marker is already on the start of the word, no left word should be returned.
+        if (startMarker == *this)
+            return { startMarker, startMarker };
+        endMarker = startMarker.nextWordEnd();
+
+        // There are two cases that will happen to indicate we are on the left word boundary:
+        // (1) The word break iterator returns the whitespace to the left of the word, and the end marker is the equivalent to the next starting word (handled by `isWhitespceAndEndsOnLeftWordBoundary`.
+        // (2) The word break iterator returns a word end before the current marker, and the next word starts at the visual equivalent to the current marker.
+
+        bool isWhitespaceAndEndsOnLeftWordBoundary = false;
+        if (equivalentTextPosition(endMarker)) {
+            auto resultRange = AXTextMarkerRange { startMarker, endMarker };
+            auto resultRangeString = resultRange.toString();
```

Comment:
Can we combine this into one line:

`auto rangeString = AXTextMarkerRange { startMarker, endMarker }.toString();`